### PR TITLE
Gobierto Data / Upgrade Perspective: 0.4.2 → 0.5.1

### PR DESCRIPTION
**Pull request by jorgeatgu**
Closes https://github.com/PopulateTools/issues/issues/1080 https://github.com/PopulateTools/issues/issues/962

## :v: What does this PR do?

Update Perspective from 0.4.2 to 0.5.1

Major change: It replace `@finos/perspective-viewer-hypegrid` which built the `table` with `canvas`, with `@finos/perspective-viewer-datagrid`, which builds the `table` with pure `HTML`.

Includes another `webpack` loader, it's necessary for the new `table`...

## :mag: How should this be manually tested?

Staging

### Before this PR

Screenshot 2020-07-14 at 15 17 15

### After this PR

### CODE REVIEW

**Comment by ferblape** [comment]
@jorgeatgu do you know why these columns from [licitaciones](https://getafe.gobify.net/datos/licitaciones/editor), which are both dates, are displayed different?

Screenshot 2020-07-16 at 06 51 32

**Comment by jorgeatgu** [comment]
@ferblape 

These days I've been rereading the Perspective documentation. I read this:

> In some cases, the inferrence algorithm may not return exactly what you'd like. For example, a column may be interpreted as a datetime when you intended it to be a string, or a column may have no values at all (yet), as it will be updated with values from a real-time data source later on.

[Perspective docs](https://perspective.finos.org/docs/md/js.html#loading-data-with-table)

So I've taken the types from each dataset, and loaded them into Perspective. Now(in local, not staging😅) it doesn't transform the dates.

Screenshot 2020-07-17 at 16 08 19


https://github.com/PopulateTools/gobierto/blob/a675dc56a327ab3c8e4be36fa5da7d310c6df509/app/javascript/gobierto_data/webapp/components/commons/Visualizations.vue#L106-L108

But there's a little problem. Perspective supports the following types:
```
integer
float
boolean
date
datetime
string
```

We have many more types, so if we pass you one of our types, the Visualization returns an error. The solution, transform our types into Perspective types https://github.com/PopulateTools/gobierto/blob/a675dc56a327ab3c8e4be36fa5da7d310c6df509/app/javascript/gobierto_data/webapp/components/commons/Visualizations.vue#L92-L102

[And, we can override Perspective settings](https://perspective.finos.org/docs/md/js.html#customizing-behavior-with-perspectiveconfigjs)

**Comment by ferblape** [comment]
Sounds a good approach to me

**Comment by ferblape** [comment]
In https://demo-datos.gobify.net/datos/datadista-ccaa-covid19-datos-isciii/editor, this query breaks perspective because can't find date column:

```sql
SELECT fecha::date as date, * 
FROM datadista_ccaa_covid19_datos_isciii 
ORDER by date DESC
LIMIT 50
```

**Comment by jorgeatgu** [comment]
@ferblape 

I added another check, and replaces `some` for `every`. Try it again

Screenshot 2020-07-24 at 09 26 20

**Review by ferblape — APPROVED** [review]

**Comment by ferblape** [comment]
ping @Crashillo

File: app/javascript/gobierto_data/webapp/components/commons/Visualizations.vue [hunk 1]
```diff
@@ -72,28 +76,79 @@ export default {
...
       }
     },
+    checkPerspectiveTypes() {
+      //Get columns generates from query
+      const arrayColumnsFromQuery = this.arrayColumnsQuery
+      //Get columns from API
+      const arrayColumnsFromAPI = Object.keys(this.objectColumns)
+
+      /* We compare the columns, if it returns true we pass the schema to Perspective, if false Perspective will convert the columns*/
+      const sameColumns = arrayColumnsFromAPI.length == arrayColumnsFromQuery.length && arrayColumnsFromAPI.every(column => arrayColumnsFromQuery.includes(column))
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/commons/Visualizations.vue]: Use three equals in comparisons

```suggestion
      const sameColumns = arrayColumnsFromAPI.length === arrayColumnsFromQuery.length && arrayColumnsFromAPI.every(column => arrayColumnsFromQuery.includes(column))
```

File: app/javascript/gobierto_data/webapp/components/commons/Visualizations.vue [hunk 2]
```diff
@@ -72,28 +76,79 @@ export default {
...
+          transformColumns[key] = 'string'
+        } else if (transformColumns[key] === 'jsonb') {
+          transformColumns[key] = 'string'
+        } else if (transformColumns[key] === 'decimal') {
+          transformColumns[key] = 'float'
+        } else if (transformColumns[key] === 'text') {
+          transformColumns[key] = 'string'
+        } else if (transformColumns[key] === 'inet') {
+          transformColumns[key] = 'integer'
+        }
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/commons/Visualizations.vue]: You should merge the clauses of the same conditions:

```
if (transformColumns[key] === 'hstore') || (transformColumns[key] === 'jsonb') ... {
   transformColumns[key] = 'string'
....
```

**Review by Crashillo — CHANGES_REQUESTED** [review]

**Review by Crashillo — APPROVED** [review]

**Pull request merged**